### PR TITLE
✨ add sautocomplete support

### PR DESCRIPTION
**Pull request by nicgirault**

### CODE REVIEW

**Comment by nicgirault** [comment]
# [Codecov](https://codecov.io/gh/lalalilo/express-sequelize-crud/pull/14?src=pr&el=h1) Report
> Merging [#14](https://codecov.io/gh/lalalilo/express-sequelize-crud/pull/14?src=pr&el=desc) into [master](https://codecov.io/gh/lalalilo/express-sequelize-crud/commit/1f1b5b863860e92b24bde8a950839c3811cf32fb&el=desc) will **decrease** coverage by `2.60%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/lalalilo/express-sequelize-crud/pull/14?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master      #14      +/-   ##
==========================================
- Coverage   91.07%   88.46%   -2.61%     
==========================================
  Files           2        2              
  Lines         112      130      +18     
  Branches       25       28       +3     
==========================================
+ Hits          102      115      +13     
- Misses          8       13       +5     
  Partials        2        2              
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/lalalilo/express-sequelize-crud/pull/14?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/lalalilo/express-sequelize-crud/pull/14?src=pr&el=footer). Last update [1f1b5b8...dcbd06c](https://codecov.io/gh/lalalilo/express-sequelize-crud/pull/14?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/index.ts [hunk 1]
```diff
@@ -100,7 +105,8 @@ export const crud = <M extends Model>(
 
 const getList = <M extends Model>(
   model: { new (): M } & typeof Model,
-  hooks?: GetListHooks
```
Comment (fTrestour) [inline src/index.ts]: Why did you switch to `Something | undefined`?

File: src/index.ts [hunk 2]
```diff
@@ -109,14 +115,9 @@ const getList = <M extends Model>(
...
-    const { count, rows } = await model.findAndCountAll({
-      offset: from,
-      limit: to - from + 1,
-      order: [sort ? JSON.parse(sort) : ['id', 'ASC']],
-      where: parsedFilter,
-      raw: true,
-    })
-
+    const { rows, count } = parsedFilter['q']
+      ? await search(parsedFilter['q'], to - from + 1, model, searchableFields)
```
Comment (fTrestour) [inline src/index.ts]: We may want to keep the sort here (or explain in the doc that search and sort can't co-exist)

**Comment by fTrestour** [comment]
image

I don't know how to read codecov reports but I guess this means you should test the throwing behaviour

File: src/index.ts (hunk 2 above)
Comment (nicgirault) [reply src/index.ts]: This does not concern our users because the react-admin dataProvider handle this alone

File: src/index.ts (hunk 1 above)
Comment (nicgirault) [reply src/index.ts]: because the param is always passed but sometime it's undefined

**Pull request merged**